### PR TITLE
Updates to use prettier for code formatting

### PR DESCRIPTION
**Pull request by shoatman**
- added prettier to root of project (fixed version intentionally to avoid folks updating to new rules until we want to universally)
- added prettier to all projects.... looks like msal-react has a dependency which uses a older version of prettier, by explicitly declaring dependency ensure we get the correct version
- ran prettier using commands created below
- Made changes post apply fixed around (moving the comments as needed):
  - @ts-ignore
  - eslint-ignore-next-line / eslint-ignore-line
 - added format:check and format:fix scripts/commands to package.json of all updated projects
 - only applied to src / test folders (where present) and ignored all files ignored in .gitignore
 - updated the eslintrc file to remove things that conflict with the default rules of prettier... intentionally using the default rules
   - Some of the things I cut were no longer recommended to be there

### CODE REVIEW

File: .eslintrc.json
Comment (tnorling) [inline .eslintrc.json:24]: Does this mean that the prettier rules will be checked alongside eslint rules in CI?

File: lib/msal-angular/src/IMsalService.ts
Comment (tnorling) [inline lib/msal-angular/src/IMsalService.ts:14]: Why the change from double to single quotes? That's going to be a ton of changes...

Comment (shoatman) [reply lib/msal-angular/src/IMsalService.ts:14]: Yes.  I didn't override any of the recommended settings from Prettier, mostly to avoid ever having to discuss the pros/cons of quote types in the future ;-)

File: .eslintrc.json
Comment (shoatman) [reply .eslintrc.json:24]: I'll double check to see what the behavior is exactly.  We made need to add an explicit format check/step to the CI process.

File: lib/msal-angular/src/IMsalService.ts
Comment (derisen) [reply lib/msal-angular/src/IMsalService.ts:14]: Perhaps we can rely on the [TypeScript repo's style guide](https://github.com/microsoft/TypeScript/wiki/Coding-guidelines)?

**Comment by tnorling** [comment]
> fixed version intentionally to avoid folks updating to new rules until we want to universally

My preference would be to set the rules we want explicitly and use ^ version rather than fixing the version and using defaults. What do other folks think?

File: lib/msal-angular/src/IMsalService.ts
Comment (shoatman) [reply lib/msal-angular/src/IMsalService.ts:14]: Here's the prettier teams opinion on options: https://prettier.io/docs/en/option-philosophy.html and Dan Abramov's issue referenced in that article: https://github.com/prettier/prettier/issues/40

**Comment by shoatman** [comment]
> > fixed version intentionally to avoid folks updating to new rules until we want to universally
> 
> My preference would be to set the rules we want explicitly and use ^ version rather than fixing the version and using defaults. What do other folks think?

See the link on options and related issue above.

File: .eslintrc.json
Comment (shoatman) [reply .eslintrc.json:24]: This is just to avoid the conflict.  Effectively ES Lint formatting errors are turned off.  Looks like we need to run prettier as a separate CI check if we want that.  Prettier does provide a package that will run prettier and then execute ES lint, but I'm be inclined to keep these separate.

**Comment by shoatman** [comment]
@peterzenz , @tnorling , @sameerag - Let me know if you want to discuss this in a meeting rather than in the PR.  I'd really like to get this in today in order to unblock a much larger PR where I don't want to have to deal with eslint formatting errors.

File: .eslintrc.json
Comment (peterzenz) [reply .eslintrc.json:24]: The prettier docs suggest a separation here to have linting rules not look at formatting, but to still check for code errors and just have prettier handle formatting/presentation.  https://prettier.io/docs/en/comparison.html

File: lib/msal-angular/src/IMsalService.ts
Comment (peterzenz) [reply lib/msal-angular/src/IMsalService.ts:14]: I mentioned in teams that this is likely worth a discussion.  I see where you're coming from here, but let's discuss the tradeoffs and any concerns if folks have any.

File: .eslintrc.json
Comment (tnorling) [reply .eslintrc.json:24]: I think having style be enforced will be important to avoid the code churn/irrelevant changes we're facing today (bonus points if the workflow can just auto commit a fix instead of failing). I don't have a strong preference about whether it's done as a separate step or as part of ES Lint.

File: lib/msal-angular/src/IMsalService.ts
Comment (tnorling) [reply lib/msal-angular/src/IMsalService.ts:14]: Sounds good, my concern is just that keeping dependencies up to date is a constant struggle and pinning dependencies makes that harder. I'm ok to try this and re-evaluate later if it becomes burdensome.

**Review by peterzenz — APPROVED** [review]

**Comment by Robbie-Microsoft** [comment]
> @peterzenz , @tnorling , @sameerag - Let me know if you want to discuss this in a meeting rather than in the PR. I'd really like to get this in today in order to unblock a much larger PR where I don't want to have to deal with eslint formatting errors.

I'd like to listen in to a meeting regarding this, if one materializes.

File: .eslintrc.json
Comment (shoatman) [reply .eslintrc.json:24]: I'll open a second PR to make the pipeline changes.  Will definitely auto-fix; however, with tests that's likely to break something (@ts-ignore) statements need to be on the line above the offending bit of code.

**Review by tnorling — APPROVED** [review]

**Comment by shoatman** [comment]
@tnorling @sameerag - Possible to get one more approval?

Thanks

shane

**Comment by shoatman** [comment]
Thanks @peterzenz @tnorling !  Very much appreciated!

**Review by konstantin-msft — APPROVED** [review]

**Pull request merged**
